services/app-api/src/resolvers/healthPlanPackage/submitHealthPlanPackage.test.ts

```diff
@@ -135,6 +139,172 @@ describe.each(flagValueTestParameters)(
+            const editedRatesPackage = await fetchTestHealthPlanPackageById(
+                server,
+                submittedEditedRates.id
+            )
+            expect(latestFormData(editedRatesPackage).rateInfos).toHaveLength(1)
+            expect(
+                latestFormData(editedRatesPackage).rateInfos[0].rateDateStart
+            ).toMatchObject(new Date(Date.UTC(2025, 1, 1)))
+            expect(
+                latestFormData(editedRatesPackage).rateInfos[0].rateDateEnd
+            ).toMatchObject(new Date(Date.UTC(2027, 1, 1)))
+            expect(
+                editedRatesPackage.revisions[0].node.submitInfo?.updatedReason
+            ).toBe('Resubmit with edited rate description')
+
+            const newRatesPackage = await fetchTestHealthPlanPackageById(
+                server,
+                submittedNewRates.id
+            )
+            expect(latestFormData(newRatesPackage).rateInfos).toHaveLength(2)
+            expect(
+                latestFormData(newRatesPackage).rateInfos[0].rateDateStart
+            ).toMatchObject(initialRateInfos().rateDateStart)
+            expect(
+                latestFormData(newRatesPackage).rateInfos[0].rateDateEnd
+            ).toMatchObject(initialRateInfos().rateDateEnd)
+            expect(
+                latestFormData(newRatesPackage).rateInfos[1].rateDateStart
+            ).toMatchObject(new Date(Date.UTC(2030, 1, 1)))
+            expect(
+                latestFormData(newRatesPackage).rateInfos[1].rateDateEnd
+            ).toMatchObject(new Date(Date.UTC(2030, 12, 1)))
+            expect(
+                newRatesPackage.revisions[0].node.submitInfo?.updatedReason
+            ).toBe('Resubmit with an additional rate added')
+
+            // also check both packages to ensure previous revision data is unchanged
+            expect(previousFormData(editedRatesPackage).rateInfos).toHaveLength(
+                1
+            )
+            expect(
+                previousFormData(editedRatesPackage).rateInfos[0].rateDateStart
+            ).toMatchObject(initialRateInfos().rateDateStart)
+            expect(
+                previousFormData(editedRatesPackage).rateInfos[0].rateDateEnd
+            ).toMatchObject(initialRateInfos().rateDateEnd)
+            expect(
+                editedRatesPackage.revisions[1].node.submitInfo?.updatedReason
+            ).toBe('Initial submission')
+
+            expect(previousFormData(newRatesPackage).rateInfos).toHaveLength(1)
+            expect(
+                previousFormData(newRatesPackage).rateInfos[0].rateDateStart
+            ).toMatchObject(initialRateInfos().rateDateStart)
+            expect(
+                previousFormData(newRatesPackage).rateInfos[0].rateDateEnd
+            ).toMatchObject(initialRateInfos().rateDateEnd)
+            expect(
+                newRatesPackage.revisions[1].node.submitInfo?.updatedReason
+            ).toBe('Initial submission')
```

Comment:
This is a great test, can we also test that indexRates pulls the right number of rates down after? Proving that we edited one rate and created another?

Reply:
@macrael Yes! I'm actively working on `indexRates` tests like this in that test file. I think indexRates tests should be much more robust.  It will be a separate PR though.